CMakeLists.txt

```diff
@@ -232,6 +216,13 @@ endfunction()
 # Set post build operations
 #
 function(mbed_set_post_build target)
+    # The mapfile name includes the executable suffix as CMake
+    # hardcoded name of the diagnostic output file for armclang it includes it.
+    # This name is ignored for the ARM toolchain.
+    # It has been added in the name here so all supported toolchain
+    # generate a mapfile with a uniform name.
+    # See https://gitlab.kitware.com/cmake/cmake/-/issues/21538
```

Comment:
it should be sufficient to link this in Armclang case below. Not certain if any comment required here?

Reply:
The mapfile name is set here. It is odd here to include `.elf.map` for all toolchain mapfiles. An explanation is needed here. No?

Reply:
> Uniform the map file name to add target + executable suffix

This should be sufficient and explain a reason why we have exec suffix in the commit message? The specific toolchain diffs should be explained in toolchain files.

Reply:
Please review [this](https://github.com/ARMmbed/mbed-os/compare/4f5528ecf95b66a3a3c92a68a100e7fd8690e407..7756dcf3786abb8083e03e9242807aad7e5fcc0e).